### PR TITLE
New isPortOpen function

### PR DESCRIPTION
**Pull request by newhoggy**
This function is for testing, and tests are included to check it actually works before use.

The `listenOn` function is defined to facilitate the testing of the test function.

### CODE REVIEW

**Comment by newhoggy** [comment]
> The code looks fine, but the commit title should mention that this function is needed for and implemented in the test. Maybe:
> 
> > tests: Implement new isPortOpen function for testing

Fixed!

File: cardano-node/test/Test/Common/Network.hs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+-- | Check if it is possible to connect to a socket address
+canConnect :: SockAddr -> IO Bool
+canConnect sockAddr = IO.bracket (IO.socket AF_INET Stream 6) IO.close' $ \sock -> do
+  result <- IO.try $ IO.connect sock sockAddr
+  case result of
+    Right () -> return True
+    Left e -> if (Errno <$> IO.ioe_errno e) == Just eCONNREFUSED
+      then return False
+      else IO.throwIO e
```
Comment (rvl) [inline cardano-node/test/Test/Common/Network.hs:39]: If you want the tests to also work on windows, then be aware that the network exceptions are different.
We have similar code in cardano-wallet:
https://github.com/input-output-hk/cardano-wallet/blob/5c1240cc668993fcb7135f244f2b1cc9c96a3238/lib/core/src/Cardano/Wallet/Network/Ports.hs#L104

Comment (newhoggy) [reply cardano-node/test/Test/Common/Network.hs]: Your suggestion worked a charm.  Thanks!

**Comment by newhoggy** [comment]
bors merge

**Pull request merged**